internal/comment/gitlab.go

```diff
@@ -0,0 +1,537 @@
+package comment
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/shurcooL/graphql"
+	"golang.org/x/oauth2"
+)
+
+// gitlabComment represents a comment on a GitLab merge request or commit. It
+// implements the Comment interface.
+type gitlabComment struct {
+	id           string
+	body         string
+	createdAt    string
+	url          string
+	discussionID string
+}
+
+// Body returns the body of the comment
+func (c *gitlabComment) Body() string {
+	return c.body
+}
+
+// Ref returns the reference to the comment. For GitLab this is a URL to the
+// HTML page of the comment.
+func (c *gitlabComment) Ref() string {
+	return c.url
+}
+
+// Less compares the comment to another comment and returns true if this
+// comment should be sorted before the other comment.
+func (c *gitlabComment) Less(other Comment) bool {
+	j := other.(*gitlabComment)
+
+	if c.createdAt != j.createdAt {
+		return c.createdAt < j.createdAt
+	}
+
+	return c.id < j.id
+}
+
+// IsHidden always returns false for GitLab since GitLab doesn't have a
+// feature for hiding comments.
+func (c *gitlabComment) IsHidden() bool {
+	return false
+}
+
+// GitLabExtra contains any extra inputs that can be passed to the GitLab comment handlers.
+type GitLabExtra struct {
+	// ServerURL is the URL of the GitLab server. This can be set to a custom URL if
+	// using GitLab enterprise. If not set, the default GitLab server URL will be used.
```

Comment:
I think it's the official name of the product, but `GitLab Enterprise` is probably okay too :)
```suggestion
	// using GitLab for Enterprise. If not set, the default GitLab server URL will be used.
```

Reply:
Doesn't seem important for a comment